e2e/internal/az/image.go

```diff
@@ -71,7 +71,12 @@ func (images Images) LatestDaily() (Image, error) {
 	}
 
 	// Reverse sort images by version
-	slices.SortFunc(dailyImages, func(i, j Image) bool { return j.Version < i.Version })
+	slices.SortFunc(dailyImages, func(i, j Image) int {
+		if natural.Less(i.Version, j.Version) {
+			return -1
+		}
+		return 1
```

Comment:
I don't think we can use `cmp` here. Since it takes the [ordered interface] as arguments for the comparison, my guess is that it would compare two strings lexicographically and we don't want that for version strings, otherwise we can get something like:
```Go
a, b := "10.0", "2.0"
fmt.Println(a < b) // This prints "true"
```
The `natural` package [handles strings differently](https://github.com/maruel/natural/blob/e76a061d070019ecd0abcd792c7273d95e227d3c/natsort.go#L22), so it's a better choice.